File: src/staleness_producer_node.cpp
#include <ros/ros.h>
#include <sensor_msgs/Image.h>
#include <sensor_msgs/image_encodings.h>

ros::Publisher producer_publisher;

static void timerCallback(const ros::TimerEvent& event)
{
    sensor_msgs::ImagePtr image_msg(new sensor_msgs::Image);
    image_msg->data.resize(640 * 480 * 3);
    image_msg->width = 640;
    image_msg->height = 480;
    image_msg->step = 640 * 3;
    image_msg->encoding = sensor_msgs::image_encodings::RGB8;
    image_msg->header.stamp = ros::Time::now();

    producer_publisher.publish(image_msg);
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "staleness_producer");
    ros::NodeHandle nh;
    ros::NodeHandle nh_priv("~");
    producer_publisher = nh_priv.advertise<sensor_msgs::Image>("image", 1);
    ros::Timer producer_timer = nh.createTimer(ros::Duration(0.01), &timerCallback);

    ros::spin();

    return 0;
}

File: scripts/staleness_producer_node.py
#!/usr/bin/env python
import rospy
from sensor_msgs.msg import Image

rospy.init_node("staleness_producer")
img_pub = rospy.Publisher("~image", Image, queue_size=1)


def timer_callback(event):
    image = Image()
    image.data = b'\0' * 640 * 480 * 3
    image.width = 640
    image.height = 480
    image.step = 640 * 3
    image.encoding = 'rgb8'
    image.header.stamp = rospy.Time.now()
    img_pub.publish(image)


timer = rospy.Timer(rospy.Duration(0.01), timer_callback)
rospy.spin()
